ir/src/main/scala/org/scalajs/core/ir/Types.scala

```diff
@@ -15,16 +15,16 @@ import Trees._
 
 object Types {
 
-  /** Type of an term (expression or statement) in the IR.
+  /** Type of a term (expression or statement) in the IR.
    *
-   *  There is a many-to-one relationship from [[ReferenceType]]s to types,
+   *  There is a many-to-one relationship from [[TypeRef]]s to `Type`s,
    *  because:
    *
    *  - `scala.Byte`, `scala.Short` and `scala.Int` collapse to [[IntType]]
```

Comment:
`AsInstanceOf(_, ClassType("scala.Short"))` is not a valid tree. It would have to be `Unbox(_, 'I') << 16 >> 16`.